packages/@aws-cdk/aws-kinesisfirehose-destinations/lib/s3.ts

```diff
@@ -0,0 +1,39 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as firehose from '@aws-cdk/aws-kinesisfirehose';
+import { CfnDeliveryStream } from '@aws-cdk/aws-kinesisfirehose';
+import * as s3 from '@aws-cdk/aws-s3';
+import { Construct } from 'constructs';
+
+/**
+ * Props for defining an S3 destination of a Kinesis Data Firehose delivery stream.
+ */
+export interface S3Props extends firehose.DestinationProps { }
+
+/**
+ * An S3 bucket destination for data from a Kinesis Data Firehose delivery stream.
+ */
+export class S3 extends firehose.DestinationBase {
```

Comment:
:) @BenChaimberg  and I went back and forth on this a lot, we thought someone might say this.  

As far as I can tell, there is only one existing example of doing this: https://github.com/aws/aws-cdk/blob/master/packages/@aws-cdk/aws-ses-actions/lib/s3.ts#L44 

I am fine with changing to `S3Destination`. It would follow from that that the Redshift and Elasticsearch classnames should be `RedshiftDestination` and `ElasticsearchDestination` then, right? 

Then as a user your code would look like:

```
new DeliveryStream(this, 'stream', {
  destinations: [ new S3Destination(bucket, { ...} ) ]
});

new DeliveryStream(this, 'stream', {
  destinations: [ new RedshiftDestination(table, { ...} ) ]
});

new DeliveryStream(this, 'stream', {
  destinations: [ new ElasticsearchDestination(domain, { ...} ) ]
});
```

Do you think that's fine?

Reply:
> ```ts
> new firehose.DeliveryStream(this, 'stream', {
>   destinations: [ new destinations.S3Destination(bucket, { ...} ) ]
> });
> 
> new firehose.DeliveryStream(this, 'stream', {
>   destinations: [ new destinations.RedshiftDestination(table, { ...} ) ]
> });
> 
> new firehose.DeliveryStream(this, 'stream', {
>   destinations: [ new destinations.ElasticsearchDestination(domain, { ...} ) ]
> });
> ```

This code makes perfect sense to me.

Reply:
With the caveat that you call the ElasticSearch one `ElasticSearchDestination` 😉.

Reply:
I like the idea of using the name of the Resource. updated the PR to go with that approach.